src/t8_schemes/t8_default/t8_default_hex/t8_default_hex_cxx.cxx

```diff
@@ -662,4 +662,69 @@ t8_default_scheme_hex_c::~t8_default_scheme_hex_c ()
    * and hence this empty function. */
 }
 
+int
+t8_default_scheme_hex_c::t8_element_pack (const t8_element_t *elements, int count, void *send_buffer, int buffer_size,
+                                          int *position, sc_MPI_Comm comm) const
+{
+  int mpiret;
+  p8est_quadrant_t *quads = (p8est_quadrant_t *) elements;
+  for (int ielem = 0; ielem < count; ielem++) {
+    mpiret = sc_MPI_Pack (&(quads[ielem].x), 1, sc_MPI_INT, send_buffer, buffer_size, position, comm);
+    SC_CHECK_MPI (mpiret);
+    mpiret = sc_MPI_Pack (&quads[ielem].y, 1, sc_MPI_INT, send_buffer, buffer_size, position, comm);
+    SC_CHECK_MPI (mpiret);
+    mpiret = sc_MPI_Pack (&quads[ielem].z, 1, sc_MPI_INT, send_buffer, buffer_size, position, comm);
+    SC_CHECK_MPI (mpiret);
+    mpiret = sc_MPI_Pack (&quads[ielem].level, 1, sc_MPI_INT8_T, send_buffer, buffer_size, position, comm);
+    SC_CHECK_MPI (mpiret);
+  }
+  return 0;
+}
+
+int
+t8_default_scheme_hex_c::t8_element_pack_size (int count, sc_MPI_Comm comm, int *pack_size) const
+{
+  int singlesize = 0;
+  int datasize = 0;
+  int mpiret;
+
+  mpiret = sc_MPI_Pack_size (1, sc_MPI_INT, comm, &datasize);
+  SC_CHECK_MPI (mpiret);
+  singlesize += datasize;
+
+  mpiret = sc_MPI_Pack_size (1, sc_MPI_INT, comm, &datasize);
+  SC_CHECK_MPI (mpiret);
+  singlesize += datasize;
+
+  mpiret = sc_MPI_Pack_size (1, sc_MPI_INT, comm, &datasize);
+  SC_CHECK_MPI (mpiret);
+  singlesize += datasize;
```

Comment:
can subsequent calls to `MPI_Pack_size` with the same argument result in different behaviour?

If not then it is better to use
` singlesize += 3*datasize;` 
instead of calling `MPI_Pack_size` 3 times.

Reply:
a quick look into the standard and i think that it is always the same:

```
[MPI_Pack_size](https://docs.open-mpi.org/en/main/man-openmpi/man3/MPI_Pack_size.3.html#mpi-pack-size) allows the application to find out how much space is needed to pack a message. A call to [MPI_Pack](https://docs.open-mpi.org/en/main/man-openmpi/man3/MPI_Pack.3.html#mpi-pack)_size(incount, datatype, comm, size) returns in size an upper bound on the increment in position that would occur in a call to MPI_Pack, with the same values for incount, datatype, and comm.

Rationale: The call returns an upper bound, rather than an exact bound, since the exact amount of space needed to pack the message may depend on the context (e.g., first message packed in a packing unit may take more space).
```

Reply:
It probably is better to add 3*datasize. I did it this way so that one can have a look at all 3 related functions next to each other and see the similar structure. (For pack and unpack the ordering is important)